routers/web/repo/helper.go

```diff
@@ -21,3 +23,11 @@ func MakeSelfOnTop(ctx *context.Context, users []*user.User) []*user.User {
 	}
 	return users
 }
+
+func HandleGitError(ctx *context.Context, msg string, err error) {
+	if git.IsErrNotExist(err) {
+		ctx.Data["NotFoundPrompt"] = ctx.Locale.Tr("repo.tree_path_not_found", ctx.Repo.TreePath, ctx.Repo.BranchName)
+		ctx.Data["NotFoundGoBackURL"] = ctx.Repo.RepoLink + "/src/branch/" + url.PathEscape(ctx.Repo.BranchName)
+	}
+	ctx.NotFoundOrServerError(msg, git.IsErrNotExist, err)
```

Comment:
Since it's clear about the error now, it could call the related functions directly:

```suggestion
	if git.IsErrNotExist(err) {
		ctx.Data["NotFoundPrompt"] = ctx.Locale.Tr("repo.tree_path_not_found", ctx.Repo.TreePath, ctx.Repo.BranchName)
		ctx.Data["NotFoundGoBackURL"] = ctx.Repo.RepoLink + "/src/branch/" + url.PathEscape(ctx.Repo.BranchName)
		ctx.NotFound(...)
	} else {
		ctx.InternalError(...)
	}
```

Reply:
Done.